packages/router/src/grpc/view-protocol-server/index-by-address.test.ts

```diff
@@ -0,0 +1,70 @@
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+import { IndexByAddressRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { ServicesInterface } from '@penumbra-zone/types';
+import { createContextValues, createHandlerContext, HandlerContext } from '@connectrpc/connect';
+import { ViewService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1alpha1/view_connect';
+import { servicesCtx } from '../../ctx';
+import { Address } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
+import { indexByAddress } from './index-by-address';
+import { getAddressByIndex, getEphemeralByIndex } from '@penumbra-zone/wasm-ts';
+
+vi.mock('@penumbra-zone/wasm-bundler', () => vi.importActual('@penumbra-zone/wasm-nodejs'));
+
+describe('IndexByAddress request handler', () => {
+  let mockServices: ServicesInterface;
+  let mockCtx: HandlerContext;
+  let testAddress: Address;
+  const testFVK =
+    'penumbrafullviewingkey1vzfytwlvq067g2kz095vn7sgcft47hga40atrg5zu2crskm6tyyjysm28qg5nth2fqmdf5n0q530jreumjlsrcxjwtfv6zdmfpe5kqsa5lg09';
+
+  beforeEach(() => {
+    mockServices = {
+      getWalletServices: () =>
+        Promise.resolve({
+          viewServer: {
+            fullViewingKey: testFVK,
+          },
+        }),
+    } as ServicesInterface;
+
+    mockCtx = createHandlerContext({
+      service: ViewService,
+      method: ViewService.methods.indexByAddress,
+      protocolName: 'mock',
+      requestMethod: 'MOCK',
+      contextValues: createContextValues().set(servicesCtx, mockServices),
+    });
+
+    testAddress = getAddressByIndex(testFVK, 0);
+  });
+
+  test('should successfully get index for address with index 0', async () => {
```

Comment:
```suggestion
  test('should successfully get index for a given address', async () => {
```

The test name should reflect _what_ you're testing, not _how_ you're testing it. `index 0` is _how_ you're testing it — i.e., you're passing it an address that happens to have an index of 0, and then asserting that the response has an index of 0. But of course, `indexByAddress` can return any account index, not just the 0th one :D